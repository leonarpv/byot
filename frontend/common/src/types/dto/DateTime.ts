import {IDateTime} from '../interfaces/IDateTime';
import moment from 'moment';

export class DateTime implements IDateTime {
  public humanReadable: string;
  public iso: string;

  constructor(dateTime?: IDateTime) {
    if (!dateTime) {
      const curr = moment();
      this.iso = curr.toISOString();
      this.humanReadable = curr.format('DD.MM.YYYY HH:ss');
    } else {
      this.iso = dateTime.iso;
      this.humanReadable = dateTime.humanReadable;
    }
  }
}
