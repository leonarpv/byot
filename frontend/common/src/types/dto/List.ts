import {IListMeta} from '../interfaces/IListMeta';
import {IList, ListEntity} from '../interfaces/IList';

export class List<T extends ListEntity> implements IList<T> {
  public entries: T[];
  public meta: IListMeta;
  constructor(list?: IList<T>);
  constructor(entries: T[], meta: IListMeta);
  constructor(entries?: T[] | IList<T>, meta?: IListMeta) {
    if (Array.isArray(entries)) {
      this.entries = entries;
      this.meta = meta!;
    } else {
      this.entries = entries?.entries || [];
      this.meta = entries?.meta || {totalCount: 0};
    }
  }
}
