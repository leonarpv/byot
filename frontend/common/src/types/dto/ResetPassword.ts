import {IResetPassword} from '../interfaces/IResetPassword';

export class ResetPassword implements IResetPassword {
  public newPassword: string;
  public passwordRepeat: string;
  public token: string;
  constructor({newPassword = '', passwordRepeat = '', token = ''}: Partial<IResetPassword> = {}) {
    this.newPassword = newPassword;
    this.passwordRepeat = passwordRepeat;
    this.token = token;
  }
}
