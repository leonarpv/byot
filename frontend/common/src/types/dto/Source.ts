import {ISource} from '../interfaces/ISource';
import {MediaType} from '../enums/MediaType';
import {SourceType} from '../enums/SourceType';

export class Source implements ISource {
  public id: string;
  public mediaType: MediaType;
  public resourceId: string;
  public sourceType: SourceType;
  public thumbnail: string;
  public url: string;
  constructor({id, mediaType, resourceId, sourceType, thumbnail, url}: ISource) {
    this.id = id;
    this.mediaType = mediaType;
    this.resourceId = resourceId;
    this.sourceType = sourceType;
    this.thumbnail = thumbnail;
    this.url = url;
  }
}
