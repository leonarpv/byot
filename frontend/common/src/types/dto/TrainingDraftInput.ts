import {ITrainingDraftInput} from '../interfaces/ITrainingDraftInput';
import {ITrainingMediaInput} from '../interfaces/ITrainingMediaInput';

export class TrainingDraftInput implements ITrainingDraftInput {
  public idTrainingSet: string;
  public label: string;
  public media: ITrainingMediaInput[];

  constructor({idTrainingSet = '', label = '', media = []}: Partial<ITrainingDraftInput> = {}) {
    this.idTrainingSet = idTrainingSet;
    this.label = label;
    this.media = media;
  }
}
