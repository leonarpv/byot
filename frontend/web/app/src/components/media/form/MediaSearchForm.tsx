import * as React from 'react';
import {Grid, IconButton, makeStyles, Theme} from '@material-ui/core';
import {AddRounded} from '@material-ui/icons';
import {MediaAutocompleteInputProps} from './MediaAutocompleteInput';
import {IMedia} from '@byot-frontend/common/src/types/interfaces/IMedia';
import {Input} from '@byot-frontend/web-common/src/components/elementary/form/Input';
import {useTranslation} from 'react-i18next';
import {CSSProperties} from '@material-ui/styles';

type AutocompleteComponent = React.ComponentType<MediaAutocompleteInputProps>;

interface Props {
  onSelect: (media: IMedia) => void;
  AutocompleteComponent: AutocompleteComponent;
}

const styles = (theme: Theme) => ({
  lastItem: {
    textAlign: 'center',
    [theme.breakpoints.up('sm')]: {
      textAlign: 'right',
    },
  } as CSSProperties,
});
const useStyles = makeStyles(styles);

let id = 0;

export const MediaSearchForm: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  const [media, setMedia] = React.useState<IMedia | undefined>();
  const {t} = useTranslation();

  const onSubmit = () => {
    if (media) {
      props.onSelect(media);
      setMedia(undefined);
    }
  };

  return (
    <Grid container spacing={2} justify="flex-end" alignItems="center">
      <Grid item xs={12} sm={6}>
        <props.AutocompleteComponent onSelect={setMedia} onClear={() => setMedia(undefined)} />
      </Grid>
      <Grid item xs={12} sm={5}>
        <Input
          label={t('Video label')}
          disabled={!media}
          value={media?.label || ''}
          onChange={e => media && setMedia({...media, label: e.target.value, id: (++id).toString()})}
        />
      </Grid>
      <Grid item xs={12} sm={1} classes={{root: styles.lastItem}}>
        <IconButton color="secondary" onClick={onSubmit}>
          <AddRounded />
        </IconButton>
      </Grid>
    </Grid>
  );
};
