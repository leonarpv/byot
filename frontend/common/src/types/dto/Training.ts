import {ITraining} from '../interfaces/ITraining';
import {IList} from '../interfaces/IList';
import {IUser} from '../interfaces/IUser';
import {IDateTime} from '../interfaces/IDateTime';
import {ITrainingSet} from '../interfaces/ITrainingSet';
import {IMedia} from '../interfaces/IMedia';
import {DateTime} from './DateTime';
import {User} from './User';
import {List} from './List';

export class Training implements ITraining {
  public id: string;
  public label: string;
  public media: IList<IMedia>;
  public owner: IUser;
  public trainingSet: ITrainingSet;
  public updatedAt?: IDateTime;
  public createdAt: IDateTime;
  constructor({id, label, media, owner, trainingSet, updatedAt, createdAt}: ITraining) {
    this.createdAt = createdAt instanceof DateTime ? createdAt : new DateTime(createdAt);
    this.updatedAt = updatedAt && updatedAt instanceof DateTime ? updatedAt : new DateTime(updatedAt);
    this.id = id;
    this.label = label;
    this.media = media instanceof List ? media : new List(media);
    this.owner = owner instanceof User ? owner : new User(owner);
    this.trainingSet = trainingSet;
  }
}
