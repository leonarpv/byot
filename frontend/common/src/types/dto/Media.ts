import {IMedia} from '../interfaces/IMedia';
import {ISource} from '../interfaces/ISource';
import {IDateTime} from '../interfaces/IDateTime';
import {DateTime} from './DateTime';
import {Source} from './Source';

export class Media implements IMedia {
  public id: string;
  public label: string;
  public source: ISource;
  public updatedAt: IDateTime;
  public createdAt?: IDateTime;

  constructor({id, label, source, updatedAt, createdAt}: IMedia) {
    this.createdAt = createdAt instanceof DateTime ? createdAt : new DateTime(createdAt);
    this.updatedAt = updatedAt && updatedAt instanceof DateTime ? updatedAt : new DateTime(updatedAt);
    this.id = id;
    this.label = label;
    this.source = source instanceof Source ? source : new Source(source);
  }
}
