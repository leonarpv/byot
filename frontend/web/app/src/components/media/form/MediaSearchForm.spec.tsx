import * as React from 'react';
import {fireEvent, render} from '@testing-library/react';
import {MediaSearchForm} from './MediaSearchForm';
import {media} from '@byot-frontend/common/test/fixtures/dto/Media';

describe('<MediaSearchForm/>', () => {
  it('should render idle state', () => {
    const {container} = render(
      <MediaSearchForm AutocompleteComponent={() => <div />} onSelect={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
  });
  it('should call event only once when the media is selected', () => {
    const onSelect = jest.fn();
    const m = media();
    const {container, getByTestId} = render(
      <MediaSearchForm
        AutocompleteComponent={({onSelect}) => (
          <div data-testid="test-media-trigger" onClick={() => onSelect(m)} />
        )}
        onSelect={onSelect}
      />
    );
    const button = getByTestId('media-form-search-button');
    fireEvent.click(button);
    expect(onSelect).not.toBeCalled();
    const autocomplete = getByTestId('test-media-trigger');
    fireEvent.click(autocomplete);
    fireEvent.click(button);
    expect(onSelect).toBeCalledWith(expect.objectContaining({...m, id: expect.any(String)}));
    expect(container).toMatchSnapshot();
  });
  it('should call event with custom label', () => {
    const onSelect = jest.fn();
    const m = media();
    const {container, getByTestId} = render(
      <MediaSearchForm
        AutocompleteComponent={({onSelect}) => (
          <div data-testid="test-media-trigger" onClick={() => onSelect(m)} />
        )}
        onSelect={onSelect}
      />
    );
    const button = getByTestId('media-form-search-button');
    const autocomplete = getByTestId('test-media-trigger');
    const labelInput = getByTestId('media-form-search-input').querySelector('input');
    expect(labelInput).not.toBeNull();
    const newLabel = 'New media label';
    fireEvent.click(autocomplete);
    fireEvent.change(labelInput!, {target: {value: newLabel}});
    fireEvent.click(button);

    expect(onSelect).toBeCalledWith(expect.objectContaining({...m, label: newLabel, id: expect.any(String)}));
    expect(container).toMatchSnapshot();
  });
});
