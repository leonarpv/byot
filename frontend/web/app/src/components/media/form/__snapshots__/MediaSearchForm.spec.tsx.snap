// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaSearchForm/> should call event only once when the media is selected 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div
        data-testid="test-media-trigger"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        data-testid="media-form-search-input"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-7 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-disabled Mui-disabled"
          data-shrink="false"
        >
          Video label
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-6 makeStyles-filledRoot-8 Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input Mui-disabled Mui-disabled"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <p
              class="MuiFormHelperText-root MuiFormHelperText-contained"
            >
              <span>
                ​
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-lastItem-5 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-1"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
        data-testid="media-form-search-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 13h-5v5c0 .55-.45 1-1 1s-1-.45-1-1v-5H6c-.55 0-1-.45-1-1s.45-1 1-1h5V6c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`<MediaSearchForm/> should call event with custom label 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div
        data-testid="test-media-trigger"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        data-testid="media-form-search-input"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-11 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-disabled Mui-disabled"
          data-shrink="false"
        >
          Video label
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-10 makeStyles-filledRoot-12 Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input Mui-disabled Mui-disabled"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <p
              class="MuiFormHelperText-root MuiFormHelperText-contained"
            >
              <span>
                ​
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-lastItem-9 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-1"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
        data-testid="media-form-search-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 13h-5v5c0 .55-.45 1-1 1s-1-.45-1-1v-5H6c-.55 0-1-.45-1-1s.45-1 1-1h5V6c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`<MediaSearchForm/> should render idle state 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    >
      <div />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5"
    >
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        data-testid="media-form-search-input"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-3 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled Mui-disabled Mui-disabled"
          data-shrink="false"
        >
          Video label
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-2 makeStyles-filledRoot-4 Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input Mui-disabled Mui-disabled"
            disabled=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <p
              class="MuiFormHelperText-root MuiFormHelperText-contained"
            >
              <span>
                ​
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-lastItem-1 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-1"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
        data-testid="media-form-search-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 13h-5v5c0 .55-.45 1-1 1s-1-.45-1-1v-5H6c-.55 0-1-.45-1-1s.45-1 1-1h5V6c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
